includes/admin/class-wc-rest-payments-refunds-controller.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Class WC_REST_Payments_Timeline_Controller
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+use WCPay\Core\Server\Request\Refund_Charge;
+use WCPay\Exceptions\API_Exception;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for the timeline, which includes all events related to an intention.
+ */
+class WC_REST_Payments_Refunds_Controller extends WC_Payments_REST_Controller {
+	/**
+	 * Endpoint path.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'payments/refund';
+
+	/**
+	 * Configure REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'process_refund' ],
+				'permission_callback' => [ $this, 'check_permission' ],
+			]
+		);
+	}
+
+	/**
+	 * Makes direct refund bypassing any order checks.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 */
+	public function process_refund( $request ) {
+		$order_id  = $request->get_param( 'order_id' );
+		$charge_id = $request->get_param( 'charge_id' );
+		$amount    = $request->get_param( 'amount' );
+		$reason    = $request->get_param( 'reason' );
+
+		if ( $order_id ) {
+			$order   = wc_get_order( $order_id );
+			$gateway = wc_get_payment_gateway_by_order( $order );
+			// TODO: Correct way to parse amount to WC style floats.
+			$result = $gateway->process_refund( $order_id, floatval( $amount / 100 ), $reason );
+
+			return rest_ensure_response( $result );
```

Comment:
I'm unsure as to how to best handle the error message. Right now we display hardcoded (in the JS) generic message regardless of the case. `$gateway->process_refund` handle some cases where we expect to return message to end user. However, other cases involve API failure and shouldn't be displayed to end user.

Reply:
We should be good as it is (except I think we miss handling cases when `$order` is empty), just ensure that we show a generic error message in UI - that'll be enough, as details likely be in order notes.